calico_mesos/calico_mesos.py

```diff
@@ -353,18 +353,10 @@ def _cleanup(hostname, container_id):
     except KeyError:
         raise IsolatorException("No endpoint found with container-id: %s" % container_id)
 
-    # Unassign any address it has.
-    for net in endpoint.ipv4_nets | endpoint.ipv6_nets:
-        assert(net.size == 1)
-        ip = net.ip
-        _log.info("Attempting to un-allocate IP %s", ip)
-        pools = datastore.get_ip_pools(ip.version)
-        for pool in pools:
-            if ip in pool:
-                # Ignore failure to unassign address, since we're not
-                # enforcing assignments strictly in datastore.py.
-                _log.info("Un-allocate IP %s from pool %s", ip, pool)
-                datastore.unassign_address(pool, ip)
+    # Release IP addresses.
+    ips = {net.ip for net in endpoint.ipv4_nets | endpoint.ipv6_nets}
+    _log.info("%s | Release IPs %s", container_id, ips)
```

Comment:
Because multiple of these processes can be running at once, the lines can get separated with other unrelated stuff.

I've added the PID now, so you can reconstruct a single call using that.  But, it might also be helpful for correlating to Mesos diags.